app/models/ems_refresh/save_inventory.rb

```diff
@@ -105,7 +107,7 @@ def save_vms_inventory(ems, hashes, target = nil)
             h[:location] = "unknown" if h[:location].blank?
 
             # build a type-specific vm or template
-            found = ems.vms_and_templates.build(h)
+            found = ems.vms_and_templates.klass.new(h)
```

Comment:
Hm I do have the 5.0.1 issue locally fixed, otherwise the benchmarks would explode :-)

In the new refresh code, I am doing only this. It seems like a normal rails behavior that ems.vms_and_templates.build is caching all the objects on ems.vms_and_templates. Or do you think it's a bug?